### PR TITLE
Fix Integration Connection Bugs

### PR DESCRIPTION
**Pull request by SanderGi**
Task: https://app.asana.com/0/1203067047205953/1207473549199484

### Q/A checklist

- [x] If you add new dependencies, did you update the lock file?
```bash
poetry lock --no-update
```
- [x] Run tests 
```bash
ulimit -n unlimited && ./scripts/run-tests.sh
```
- [x] Do a self code review of the changes - Read the diff at least twice.  
- [x] Carefully think about the stuff that might break because of this change - this sounds obvious but it's easy to forget to do "Go to references" on each function you're changing and see if it's used in a way you didn't expect. 
- [x] The relevant pages still run when you press submit
- [x] The API for those pages still work (API tab)
- [x] The public API interface doesn't change if you didn't want it to (check API tab > docs page)
- [x] Do your UI changes (if applicable) look acceptable on mobile?
- [x] Ensure you have not regressed the import time unless you have a good reason to do so. 
You can visualize this using tuna:
```bash
python3 -X importtime -c 'import server' 2> out.log && tuna out.log
```
To measure import time for a specific library:
```bash
$ time python -c 'import pandas'

________________________________________________________
Executed in    1.15 secs    fish           external
   usr time    2.22 secs   86.00 micros    2.22 secs
   sys time    0.72 secs  613.00 micros    0.72 secs
```
To reduce import times, import libraries that take a long time inside the functions that use them instead of at the top of the file:
```python
def my_function():
    import pandas as pd
    ...
```

### CODE REVIEW

File: routers/slack_api.py [hunk 1]
```diff
@@ -140,12 +157,12 @@ def slack_connect_redirect(request: Request):
             BotIntegration.objects.filter(pk=bi.pk).update(**config)
             bi.refresh_from_db()
 
+    redirect = connect(bi, current_run, published_run)
```
Comment (devxpy) [inline routers/slack_api.py]: This broke the creation of personal channels as it always set slack_create_personal_channels=False

File: routers/facebook_api.py [hunk 2]
```diff
@@ -146,15 +153,31 @@ def fb_connect_redirect(request: Request):
             status_code=400,
         )
 
+    # only connect to pages that are not already connected
```
Comment (devxpy) [inline routers/facebook_api.py:156]: Any reason for doing this?

Comment (devxpy) [reply routers/facebook_api.py:156]: reverting for now as it seems to keep pages that are disconnected from facebook around as integrations

Comment (devxpy) [reply routers/facebook_api.py]: Ok yeah dont wanna loose the analytics for those integrations, fixed in https://github.com/GooeyAI/gooey-server/commit/c08a3c6ad00d7445804b456356515a6edf906195

File: routers/slack_api.py (hunk 1 above)
Comment (SanderGi) [reply routers/slack_api.py]: I believe sean requested the creation of personal channels disabled

Comment (SanderGi) [reply routers/slack_api.py]: image

Comment (SanderGi) [reply routers/slack_api.py]: I did go through and test all the integrations. Would be good to have some sort of automated test for this too, even if it mocks the 3rd party portions

Comment (SanderGi) [reply routers/slack_api.py]: Tracking here: https://app.asana.com/0/1203067047205953/1204911106314617

Comment (devxpy) [reply routers/slack_api.py]: I am not sure how we enable creation of personal channels in this case. There would be no way for end users to do it, so maybe we need some ux around this too

**Pull request merged**